test/functional/fundrawtransaction.py

```diff
@@ -53,6 +53,11 @@ def run_test(self):
         self.nodes[0].generate(121)
         self.sync_all()
 
+        # ensure that setting changPosition in fundraw with an exact match is handled properly
+        rawmatch = self.nodes[2].createrawtransaction([], {self.nodes[2].getnewaddress():50})
+        rawmatch = self.nodes[2].fundrawtransaction(rawmatch, {"changePosition":1, "subtractFeeFromOutputs":[0]})
+        assert_equal(rawmatch["changepos"], -1)
```

Comment:
if no resulting change exists, it will be -1 regardless. It was just broken before in the exact match case.